fbw-a32nx/src/wasm/systems/a320_systems/src/payload/mod.rs

```diff
@@ -0,0 +1,663 @@
+            let pax_num = 0;
+            if self.pax_num(ps) == pax_num {
+                return true;
+            }
+        }
+        false
+    }
+
+    fn board_rate(&self) -> BoardingRate {
+        self.board_rate
+    }
+
+    fn pax_num(&self, ps: A320Pax) -> i8 {
+        self.pax[ps as usize].pax_num() as i8
+    }
+
+    fn total_pax_num(&self) -> i32 {
+        let mut pax_num = 0;
+        for ps in A320Pax::iterator() {
+            pax_num += self.pax_num(ps) as i32;
+        }
+        pax_num
+    }
+
+    fn total_max_pax(&self) -> i32 {
+        let mut max_pax = 0;
+        for ps in A320Pax::iterator() {
+            max_pax += A320_PAX[ps].max_pax as i32;
+        }
+        max_pax
+    }
+
+    fn pax_target_num(&self, ps: A320Pax) -> i8 {
+        self.pax[ps as usize].pax_target_num() as i8
+    }
+
+    fn pax_is_sync(&mut self, ps: A320Pax) -> bool {
+        self.pax[ps as usize].payload_is_sync()
+    }
+
+    fn pax_is_target(&self, ps: A320Pax) -> bool {
+        self.pax[ps as usize].pax_is_target()
+    }
+
+    fn sync_pax(&mut self, ps: A320Pax) {
+        self.pax[ps as usize].load_payload();
+    }
+
+    fn move_all_pax(&mut self, ps: A320Pax) {
+        self.pax[ps as usize].move_all_pax();
+    }
+
+    fn move_one_pax(&mut self, ps: A320Pax) {
+        self.pax[ps as usize].move_one_pax();
+    }
+
+    fn reset_pax_and_target(&mut self, ps: A320Pax) {
+        self.pax[ps as usize].reset_pax_and_target();
+    }
+
+    fn reset_cargo_and_target(&mut self, cs: A320Cargo) {
+        self.cargo[cs as usize].reset_cargo_and_target();
+    }
+
+    fn cargo_is_sync(&mut self, cs: A320Cargo) -> bool {
+        self.cargo[cs as usize].payload_is_sync()
+    }
+
+    fn cargo_is_target(&self, cs: A320Cargo) -> bool {
+        self.cargo[cs as usize].cargo_is_target()
+    }
+
+    fn move_all_cargo(&mut self, cs: A320Cargo) {
+        self.cargo[cs as usize].move_all_cargo();
+    }
+
+    fn move_one_cargo(&mut self, cs: A320Cargo) {
+        self.cargo[cs as usize].move_one_cargo();
+    }
+
+    fn load_all_cargo_percent(&mut self, percent: f64) {
+        for cs in A320Cargo::iterator() {
+            self.load_cargo_percent(cs, percent);
+        }
+    }
+
+    fn load_cargo_percent(&mut self, cs: A320Cargo, percent: f64) {
+        self.cargo[cs as usize].load_cargo_percent(percent)
+    }
+
+    fn sync_cargo(&mut self, cs: A320Cargo) {
+        self.cargo[cs as usize].load_payload();
+    }
+
+    fn is_boarding(&self) -> bool {
+        self.is_boarding
+    }
+
+    fn is_gsx_enabled(&self) -> bool {
+        self.is_gsx_enabled
+    }
+
+    fn stop_boarding(&mut self) {
+        self.is_boarding = false;
+    }
+
+    fn per_pax_weight(&self) -> Mass {
+        self.per_pax_weight.get()
+    }
+}
+impl SimulationElement for A320Payload {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        for ps in 0..self.pax.len() {
+            self.pax[ps].accept(visitor);
```

Comment:
You can use the `accept_iterable!` macro instead.

```suggestion
        accept_iterable!(self.pax, visitor);
        accept_iterable!(self.cargo, visitor);
```